### PR TITLE
Leak example

### PR DESCRIPTION
**Pull request by guizmaii**
I implemented an example of a leak I want to avoid in order to help me work on the solution.

Here is the response the `leaky` endpoint produces:
```
{"name":"BadRequest","detail":"This message should not leak to the public"}
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,3 +11,70 @@ bin
 .factorypath
 .settings
 logs
+### Scala template
+*.class
```
Comment (ignasi35) [inline .gitignore:15]: Isn't `target` enough to ignore all `.class`?

File: .gitignore [hunk 2]
```diff
@@ -11,3 +11,70 @@ bin
...
+.idea/**/dataSources.ids
+.idea/**/dataSources.xml
+.idea/**/dataSources.local.xml
+.idea/**/sqlDataSources.xml
+.idea/**/dynamic.xml
+.idea/**/uiDesigner.xml
+
+# Gradle:
+.idea/**/gradle.xml
+.idea/**/libraries
```
Comment (ignasi35) [inline .gitignore:37]: I think `.idea` (without `/` or `*`)  does the trick here.

File: exception-examples-api/src/main/scala/com/example/hello/api/ExceptionExamplesService.scala [hunk 3]
```diff
@@ -18,13 +19,16 @@ trait ExceptionExamplesService extends Service {
 
   def hello(id: String): ServiceCall[NotUsed, String]
 
+  def leaky: ServiceCall[NotUsed, String]
```
Comment (ignasi35) [inline exception-examples-api/src/main/scala/com/example/hello/api/ExceptionExamplesService.scala:22]: prefer `disclose` over `leak`

File: exception-examples-impl/src/main/scala/com/example/hello/impl/ExceptionExamplesServiceImpl.scala [hunk 4]
```diff
@@ -1,14 +1,29 @@
...
 
+  /**
+    * TO TEST:
+    * --------
+    *
+    * - in a console, do:             `sbt clean docker:publishLocal`
+    * - launch the generated script:  `./exception-examples-impl/target/docker/stage/opt/docker/bin/exception-examples-impl`
+    * - in another console:           `curl http://localhost:9000/api/leaky`
+    */
+  override def leaky: ServiceCall[NotUsed, String] = ServiceCall { _ =>
```
Comment (ignasi35) [inline exception-examples-impl/src/main/scala/com/example/hello/impl/ExceptionExamplesServiceImpl.scala:24]: 👍🏼 Nice docs! How about copy/pasting to `README.md` so we increase exposure to them? (so, keep the comment here too)

File: exception-examples-impl/src/main/scala/com/example/hello/impl/ExceptionExamplesServiceImpl.scala [hunk 5]
```diff
@@ -1,14 +1,29 @@
...
+    val cause = new RuntimeException(s"This message should not leak to the public")
```
Comment (ignasi35) [inline exception-examples-impl/src/main/scala/com/example/hello/impl/ExceptionExamplesServiceImpl.scala:25]: Prefer `disclose` over `leak`?

**Review by ignasi35 — COMMENTED** [review]
I think `leak` is not the right word. I know it's from Lagom docs but I think we should change it there too. IMHO `disclose` is better in this context.

**Comment by ignasi35** [comment]
Merging as is, we can discuss leak-vs-disclose on a separate ticket.

File: .gitignore (hunk 1 above)
Comment (guizmaii) [reply .gitignore:15]: The file is generated by a plugin in IntelliJ. I never red it ^^

File: .gitignore (hunk 2 above)
Comment (guizmaii) [reply .gitignore:37]: Same answer than above ;)

File: exception-examples-api/src/main/scala/com/example/hello/api/ExceptionExamplesService.scala (hunk 3 above)
Comment (guizmaii) [reply exception-examples-api/src/main/scala/com/example/hello/api/ExceptionExamplesService.scala:22]: Ok. I'll change

File: exception-examples-impl/src/main/scala/com/example/hello/impl/ExceptionExamplesServiceImpl.scala (hunk 4 above)
Comment (guizmaii) [reply exception-examples-impl/src/main/scala/com/example/hello/impl/ExceptionExamplesServiceImpl.scala:24]: Ok

**Pull request merged**